src/vm/moar/QAST/QASTCompilerMAST.nqp

```diff
@@ -507,7 +507,7 @@ my class MASTCompilerInstance {
                 if $got == $MVM_reg_int64 {
                     %core_op_generators{'coerce_is'}($res_reg, $reg);
                 }
-                elsif $got == $MVM_reg_int32 {
+                elsif $got == $MVM_reg_int32 || $got == $MVM_reg_int16 || $got == $MVM_reg_int8 || $got == $MVM_reg_uint32 || $got == $MVM_reg_uint16 || $got == $MVM_reg_uint8 {
```

Comment:
This is probably correct but a bit confusing. Took me a second read to see that since we're coercing to int64 first, the coerce_is is correct (as opposed to a coerce_us like for MVM_reg_uint64). And it's correct, since we're handling only types that are smaller than int64, so it can't overflow.